app/src/main/java/com/globant/weatherapp/activities/WeatherInfoActivity.kt

```diff
@@ -0,0 +1,46 @@
+package com.globant.weatherapp.activities
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.globant.weatherapp.R
+import com.globant.weatherapp.utils.DataSource
+import com.globant.weatherapp.utils.WeatherRecyclerViewAdapter
+import kotlinx.android.synthetic.main.activity_weather_info_layout.recycler_view
+
+class WeatherInfoActivity : AppCompatActivity() {
+
+    private lateinit var weatherAdapter: WeatherRecyclerViewAdapter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_weather_info_layout)
+
+        val cityId = intent.extras.getSerializable(CITY_ID)
+
+        initRecyclerView()
+        addDataSet()
+    }
+
+    private fun initRecyclerView() {
+        recycler_view.apply {
+            layoutManager = LinearLayoutManager(this.context)
+            weatherAdapter = WeatherRecyclerViewAdapter()
+            adapter = weatherAdapter
+        }
+    }
+
+    private fun addDataSet() {
```

Comment:
this method must be in the view and called from the presenter

Reply:
Yes! Done!